framework/src/vectorpostprocessors/PointSamplerBase.C

```diff
@@ -162,7 +171,18 @@ PointSamplerBase::getLocalElemContainingPoint(const Point & p)
   const Elem * elem = (*_pl)(p);
 
   if (elem && elem->processor_id() == processor_id())
+  {
+    // Print a warning if it's on a face and a variable is discontinuous
+    if (_warn_discontinuous_face_values && _discontinuous_variables)
+      for (const auto s : elem->side_index_range())
+        if (elem->build_side_ptr(s)->contains_point(p))
```

Comment:
This operation could be quite slow, is point locator at Line 171 already doing something and changing its status that we can quarry after Line 171?

Reply:
I'm fine with that. My original concern was the _double_ locator call; in reality, we should call `operator()` for continuous and `locate_node` for discontinuous.

I'd also like to add that calling contains point on sides is a lot cheaper than on volumes, like we do within the point locator :). But regardless - using the locator for all of it is the better move here.

Reply:
I'd say we still want () for discontinuous, it's not because the variable isnt sampled at a node that we dont care about the value. 

My concern with the new approach is whether the set of elements is consistently in the same order across processes. If not arbitrarily taking the first one in the set isnt ok. @jwpeterson do you know if that is true?

Reply:
I still don't fully agree with this - you don't have to be at a node for the discontinuity to be an issue.

Reply:
You are right there is the question of what to do on the outer boundary. The value is ill-defined, it certainly wont return the DirichletBC for example. 

This new approach won't capture that

Reply:
Oh, duh. I keep forgetting that we're talking about the two `operator()` checks, not `locate_node`.

For continuous: use as it was before.

For discontinuous: use the other operator. Create a `std::set<dof_id_type>` of element IDs from the returned `std::set<const Elem *>`. Call `comm().set_union()` on that set. If the size of the set is > 1, show a warning. Return nullptr for all processors except for the one that _owns_ the element with the lowest ID, and return the pointer to said element.

This will enable deterministic behavior with problems with the same partitioning.